docs/tutorials/sinks/2_http_sink.md

```diff
@@ -0,0 +1,504 @@
+    }
+```
+
+# Request Builder
+
+Next we creata request builder that turns the event into a request. The 
+request is used by a Tower service that is responsible for actually sending
+the data.
+
+The request looks like:
+
+```rust
+#[derive(Clone)]
+struct BasicRequest {
+    payload: Bytes,
+    finalizers: EventFinalizers,
+    metadata: RequestMetadata,
+}
+```
+
+The fields in the request are:
+
+## payload
+
+The payload is the actual bytes that we will be sending out. These are the bytes
+generated by our `BasicEncoder`.
+
+## finalizers
+
+`EventFinalizers` is a collection of `Finalizers`. A `Finalizer` is used to track 
+the status of a given event and is used to support 
+[end to end acknowledgements](https://vector.dev/docs/about/under-the-hood/guarantees/#acknowledgement-guarantees).
+
+## metadata
+
+The metadata contains additional data that is used to emit various metrics when 
+a request is successfully sent.
+
+
+We need to implement a number of traits for the request to access these fields:
+
+```rust
+impl MetaDescriptive for BasicRequest {
+    fn get_metadata(&self) -> RequestMetadata {
+        self.metadata
+    }
+}
+
+impl Finalizable for BasicRequest {
+    fn take_finalizers(&mut self) -> EventFinalizers {
+        self.finalizers.take_finalizers()
+    }
+}
+```
+
+The request builder must implement the `RequestBuilder<>` trait: 
+
+```rust
+impl RequestBuilder<Event> for BasicRequestBuilder {
+```
+
+There are a number of stages in the request builder process -  first the input 
+is split out into metadata and actual event data. The event data is encoded. 
+The results from the encoding are are passed with the metadata to create the 
+final request that is passed to the `Tower` sink.
+
+Here, the trait is generic over `Event` which is the input type that is 
+passed in to start the request building process.
+
+There are a number of associated types:
+
+```rust
+    type Metadata = EventFinalizers;
+    type Events = Event;
+    type Encoder = BasicEncoder;
+    type Payload = Bytes;
+    type Request = BasicRequest;
+    type Error = std::io::Error;
+```
+
+## Metadata
+
+`Metadata` is the type for any information to be passed while building the request
+that is additional to the actual event being used. In this case we just need the
+`EventFinalizers`.
+
+## Events
+
+Events contains the type that is the event type passed to the `Encoder`.
+
+## Encoder
+
+The `Encoder` is the type that is used to encode the event to create the final
+payload. We are using the `BasicEncoder` described earlier.
+
+## Payload
+
+The `Payload` is the type of the final data that is encoded.
+
+## Request
+
+`Request` is the type that is sent to the final service. This is the `BasicRequest`
+we described earlier.
+
+## Error
+
+`Error` is the type of any errors that are creating while encoding the event.
+
+
+The following functions for the `RequestBuilder` trait need implementing:
+
+## compression
```

Comment:
The transition from the prior section to this one is jarring.
The sentence on line 235 appears to be part of the Error section.